File: ai/src/main/scala/org/pingel/gestalt/core/refactor/RenameFormRefactoring.java
package org.pingel.gestalt.core.refactor;

import org.pingel.gestalt.core.Lexicon;
import org.pingel.gestalt.core.Name;

public class RenameFormRefactoring extends Refactoring {

	Name from;
	Name to;
	
	public RenameFormRefactoring(Lexicon lexicon, Name from, Name to)
	{
		super(lexicon);
		this.from = from;
		this.to = to;
	}

	public void execute()
	{
		lexicon.renameForm(from, to);
	}
	
	
}

File: ai/src/main/scala/org/pingel/gestalt/core/refactor/RenameFormRefactoring.scala
package org.pingel.gestalt.core.refactor

import org.pingel.gestalt.core.Lexicon
import org.pingel.gestalt.core.Name

class RenameFormRefactoring(lexicon: Lexicon, from: Name, to: Name)
extends Refactoring(lexicon)
{
	def execute(): Unit = {
		lexicon.renameForm(from, to)
	}
}
